receiver/podmanreceiver/receiver.go

```diff
@@ -57,11 +66,73 @@ func newReceiver(
 		set:           set,
 	}
 
-	scrp, err := scraperhelper.NewScraper(typeStr, recv.scrape, scraperhelper.WithStart(recv.start))
+	return recv, err
+}
+
+func (r *receiver) registerMetricsConsumer(mc consumer.Metrics, set component.ReceiverCreateSettings) error {
+	r.metricsConsumer = mc
+	scrp, err := scraperhelper.NewScraper(typeStr, r.scrape, scraperhelper.WithStart(r.start))
 	if err != nil {
-		return nil, err
+		return err
+	}
+	r.metricsComponent, err = scraperhelper.NewScraperControllerReceiver(&r.config.ScraperControllerSettings, set, mc, scraperhelper.AddScraper(scrp))
+	return err
+}
+
+func (r *receiver) registerLogsConsumer(lc consumer.Logs) {
+	r.logsConsumer = lc
+}
+
+func (r *receiver) Start(ctx context.Context, host component.Host) error {
+	// Check for logs pipeline
+	if r.logsConsumer == nil {
+		r.set.Logger.Warn("Logs receiver is not set")
+	} else {
+		eventBackoff := backoff.NewExponentialBackOff()
+		eventBackoff.InitialInterval = 2 * time.Second
+		eventBackoff.MaxInterval = 10 * time.Minute
+		eventBackoff.Multiplier = 2
+		eventBackoff.MaxElapsedTime = 0
+		r.isLogsShutdown = false
+		go func() {
```

Comment:
Perhaps you need to cleanly return from this go routine during the shutdown?

Reply:
I didn't completely understand what you meant by cleanly return
But I am assuming you meant to say that when the shutdown happens, we should have a proper log message. If yes, in the case of a shutdown, logs are added by the parent process and we wouldn't have to add it separately.

To explain further:
`errorWhileRetry` has two cases:  1. nil error 2. Non-nil error.
-  In the non-nil error case, we are adding the log.
- Nil error will only occur when the shutdown will be called. And when the shutdown happens, a log will be added by the parent process, that's why we didn't add the log.

correct me if I am wrong.

Reply:
Sorry, I should have expanded a bit. I meant that when shutdown happens, this go func will still exist and will only exit when the backoff retry has been exhausted. This means that you could potentially still be waiting for a retry when the component shuts down, so, your handleEvents will be called after the component stopped. This should *not* happen.

Reply:
Got it. To resolve the issue where goroutine was `waiting for a retry when the component shuts down`, we have used a cancellable context(which will make sure that when shutdown happens, the request will be closed and the goroutine will gracefully exit) and passed it to the podman request (in `getEventsResponse`), and handled the cases in respective functions.
Also, we have removed `isLogsShutdown` as it's not being used now.